delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutesSpec.scala

```diff
@@ -346,6 +349,51 @@ class ResourcesRoutesSpec extends BaseRouteSpec with IOFromMap with CatsIOValues
       }
     }
 
+    "fail to update a schema without resources/write permission" in {
+      aclCheck.subtract(AclAddress.Root, Anonymous -> Set(resources.write)).accepted
+      Put(s"/v1/resources/$projectRef/_/someId/updateSchema") ~> routes ~> check {
+        response.status shouldEqual StatusCodes.Forbidden
+        response.asJson shouldEqual jsonContentOf("errors/authorization-failed.json")
+      }
+    }
+
+    "fail to update a schema when providing a non-existing schema" in {
+      aclCheck.append(AclAddress.Root, Anonymous -> Set(resources.write)).accepted
+      thereIsAResourceWithSchema { id =>
+        Put(s"/v1/resources/$projectRef/wrongSchema/$id/updateSchema") ~> routes ~> check {
+          response.status shouldEqual StatusCodes.NotFound
+          response.asJson.hcursor.get[String]("@type").toOption should contain("InvalidSchemaRejection")
+        }
+      }
+    }
+
+    "fail to update schema when providing _ as schema" in {
+      thereIsAResourceWithSchema { id =>
+        Put(s"/v1/resources/$projectRef/_/$id/updateSchema") ~> routes ~> check {
+          response.status shouldEqual StatusCodes.BadRequest
+          response.asJson.hcursor.get[String]("@type").toOption should contain("IdenticalSchema")
+        }
+      }
+    }
+
+    "fail to update schema when providing current schema" in {
+      thereIsAResourceWithSchema { id =>
+        Put(s"/v1/resources/$projectRef/myschema/$id/updateSchema") ~> routes ~> check {
+          response.status shouldEqual StatusCodes.BadRequest
+          response.asJson.hcursor.get[String]("@type").toOption should contain("IdenticalSchema")
+        }
+      }
+    }
+
+    "succeed to update the schema" in {
+      thereIsAResourceWithSchema { id =>
```

Comment:
I think both this and the previous test would read better if you pass the schema name to `thereIsAResourceWithSchema`

Reply:
I was thinking about what I did in another test class when I read your comment. Pushing changes now